### PR TITLE
Mod 9610 status of funds hover modal

### PR DESCRIPTION
**Pull request by chasls2**
**High level description:**

Added some wording to a tooltip that changes based on what status of fund level its on

**Technical details:**

Hover modal on each level of the Status of Funds chart includes the “Click bar to view [next funding level]”

“Click bar…” text is updated to the following in descending order: TBD

Click a sub-component to view Federal Accounts

Click a Federal Account to view Treasury Accounts

Click a Treasury Account to view Program Activities or Object Classes

**JIRA Ticket:**
[DEV-9610](https://federal-spending-transparency.atlassian.net/browse/DEV-9610)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [ ] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [ ] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [ ] Design review complete `if applicable`
- [ ] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

**Comment by nick-torres** [comment]
image
space on here looks a little big?

**Comment by chasls2** [comment]
> image space on here looks a little big?

it is.. I had to change the height of the tooltip because the tooltip itself was blocking the mouse from interacting with the chart. Since the tooltip i based on where the mouse is it was always blocking if the title was too long.

**Comment by nick-torres** [comment]
can't you make the tooltip height back to what it was and just play with the transform mouse locations?

**Comment by chasls2** [comment]
That's what I originally did and the space between the mouse and tooltip looks big instead.

**Comment by chasls2** [comment]
Yeah, I know that number 270px on the y is as close as I can get to looking normal. made the change

File: src/js/components/agency/visualizations/StatusOfFundsChart.jsx [hunk 1]
```diff
@@ -1005,10 +1015,10 @@ const StatusOfFundsChart = ({
                     width={288}
                     styles={!toggle ? {
                         position: 'absolute',
-                        transform: `translate(${mouseValue.x - 144}px,${mouseValue.y - 230}px)`
+                        transform: `translate(${mouseValue.x - 144}px,${mouseValue.y - 270}px)`
```
Comment (nick-torres) [inline src/js/components/agency/visualizations/StatusOfFundsChart.jsx:1018]: isHovered &&
                <TooltipWrapper
                    className="sof_chart-tt"
                    width={288}
                    styles={!toggle ? {
                        position: 'absolute',
                        transform: `translate(${mouseValue.x - 144}px,${mouseValue.y - 230}px)`
the above is a snippet obviously, but produces the below image for me, does it not look like that for you??
image

Comment (chasls2) [reply src/js/components/agency/visualizations/StatusOfFundsChart.jsx:1018]: It does until I click into the next level
Then it changes and does the block I was mentioning before hand

Comment (nick-torres) [reply src/js/components/agency/visualizations/StatusOfFundsChart.jsx:1018]: got it, looked at it more and what you'll have to do is based on the level (which is a prop you have access to) and based on the level you'll have to return a slightly different number if you're at a level greater than 0
obviously you know ${mouseValue.y - 230}px works fine for the default case 

you'll also have to check how it looks when the toggle is also enabled, let me know if that doesn't make sense

File: src/js/components/agency/visualizations/StatusOfFundsChart.jsx [hunk 2]
```diff
@@ -170,6 +170,19 @@ const StatusOfFundsChart = ({
...
+        if (level === 0) {
+            return <><hr /><div className="tooltip__text-note">Click a sub-component to view <br />Federal Accounts</div></>;
+        } else if (level === 1) {
+            return <><hr /><div className="tooltip__text-note">Click a Federal Account to view <br />Treasury Accounts</div></>;
+        } else if (level === 2) {
+            return <><hr /><div className="tooltip__text-note">Click a Treasury Account to view <br />Program Activities or Object Classes</div></>;
+        }
+        return null;
+    };
+    const tooltipHeight = level === 1 ? 280 : 230;
```
Comment (nick-torres) [inline src/js/components/agency/visualizations/StatusOfFundsChart.jsx:183]: do these need to say === 1 or >=1?

Comment (chasls2) [reply src/js/components/agency/visualizations/StatusOfFundsChart.jsx:183]: I find that the longest titles are in level 1. and drop back down to normal after 1. So I did ===1.

**Review by nick-torres — APPROVED** [review]

**Pull request merged**